### PR TITLE
Replace Mailgun by Sendgrid since we are using the last, add link to service and it's privacy policy.

### PR DESCRIPTION
**Pull request by cabello**
- Since we are using Sendgrid's API, I replaced Mailgun by Sendgrid.
- I also added a link to Sendgrid home page and privacy policy.

### CODE REVIEW

File: formspree/templates/static_pages/index.html [hunk 1]
```diff
@@ -98,7 +98,7 @@ <h3 class="light">You don't even have to register.</h3>
             <div class="col-1-2">
                 <div class="card">
                     <h4>What about privacy?</h4>
-                    <p>We don't store contents of the form submissions. Emails are sent using Mailgun's API, so on that end their privacy policies apply.</p>
+                    <p>We don't store contents of the form submissions. Emails are sent using <a href="https://sendgrid.com">Sendgrid's</a> API, so on that end their <a href="https://sendgrid.com/privacy">privacy policies</a> apply.</p>
```
Comment (fiatjaf) [inline formspree/templates/static_pages/index.html]: In fact we do store. I should have changed this line.

Comment (cabello) [inline formspree/templates/static_pages/index.html]: I wasn't aware that we are storing the contents. Do you want me to update that?

**Comment by cabello** [comment]
Do you want me to reissue this updated just mentioning sendgrid?

**Comment by fiatjaf** [comment]
No, thank you. There's no need for this now.

**Pull request closed without merging**